packages/plugin-commands-licenses/src/licenses.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  FILTERING,
+  OPTIONS,
+  UNIVERSAL_OPTIONS,
+} from '@pnpm/common-cli-options-help'
+import { Config, types as allTypes } from '@pnpm/config'
+import { PnpmError } from '@pnpm/error'
+import { licences } from '@pnpm/licenses'
+import pick from 'ramda/src/pick'
+import renderHelp from 'render-help'
+import { renderLicences } from './outputRenderer'
+
+export function rcOptionsTypes () {
+  return {
+    ...pick(
+      ['dev', 'global-dir', 'global', 'json', 'long', 'optional', 'production'],
+      allTypes
+    ),
+    compatible: Boolean,
+    table: Boolean,
+  }
+}
+
+export const cliOptionsTypes = () => ({
+  ...rcOptionsTypes(),
+  recursive: Boolean,
+})
+
+export const shorthands = {
+  D: '--dev',
+  P: '--production',
+}
+
+export const commandNames = ['licenses']
+
+export function help () {
+  return renderHelp({
+    description: `Check for licenses packages. The check can be limited to a subset of the installed packages by providing arguments (patterns are supported).
+
+Examples:
+pnpm licenses
+pnpm licenses --long
+pnpm licenses gulp-* @babel/core`,
+    descriptionLists: [
+      {
+        title: 'Options',
+
+        list: [
+          {
+            description:
+              'By default, details about the packages (such as a link to the repo) are not displayed. \
+To display the details, pass this option.',
+            name: '--long',
+          },
+          {
+            description: 'Show information in JSON format',
+            name: '--json',
+          },
+          {
+            description: 'Check only "dependencies" and "optionalDependencies"',
+            name: '--prod',
+            shortAlias: '-P',
+          },
+          {
+            description: 'Check only "devDependencies"',
+            name: '--dev',
+            shortAlias: '-D',
+          },
+          {
+            description: 'Don\'t check "optionalDependencies"',
+            name: '--no-optional',
+          },
+          OPTIONS.globalDir,
+          ...UNIVERSAL_OPTIONS,
+        ],
+      },
+      FILTERING,
+    ],
+    url: docsUrl('licenses'),
+    usages: ['pnpm licenses [options]'],
+  })
+}
+
+export const completion: CompletionFunc = async (cliOpts) => {
+  return readDepNameCompletions(cliOpts.dir as string)
+}
+
+export type LicensesCommandOptions = {
+  compatible?: boolean
+  long?: boolean
+  recursive?: boolean
+  json?: boolean
+} & Pick<
+Config,
+| 'allProjects'
+| 'ca'
+| 'cacheDir'
+| 'cert'
+| 'dev'
+| 'dir'
+| 'engineStrict'
+| 'fetchRetries'
+| 'fetchRetryFactor'
+| 'fetchRetryMaxtimeout'
+| 'fetchRetryMintimeout'
+| 'fetchTimeout'
+| 'global'
+| 'httpProxy'
+| 'httpsProxy'
+| 'key'
+| 'localAddress'
+| 'lockfileDir'
+| 'networkConcurrency'
+| 'noProxy'
+| 'offline'
+| 'optional'
+| 'production'
+| 'rawConfig'
+| 'registries'
+| 'selectedProjectsGraph'
+| 'strictSsl'
+| 'tag'
+| 'userAgent'
+| 'storeDir'
+| 'virtualStoreDir'
+| 'modulesDir'
+| 'pnpmHomeDir'
+> &
```

Comment:
I am not sure any of these settings are needed for the license command. These settings are needed for the install command. This command is not installing anything, so settings related to networking, proxies, etc are not needed.